File: src/main/java/cz/stasimek/fakturaceeasypeasy/util/AppStringUtils.java
package cz.stasimek.fakturaceeasypeasy.util;

import org.apache.commons.lang3.StringUtils;


public class AppStringUtils {

	private AppStringUtils() {
	}

	public static String camelCaseToWords(String text) {
		if (StringUtils.isEmpty(text)) {
			return text;
		}
		String firstLetter = text.substring(0, 1).toLowerCase();
		String restLetters = text.substring(1).replaceAll("([A-Z])", " $1").toLowerCase();
		return firstLetter + restLetters;
	}
}

File: src/main/kotlin/cz/stasimek/fakturaceeasypeasy/util/AppStringUtils.kt
package cz.stasimek.fakturaceeasypeasy.util

import java.util.*

object AppStringUtils {

    fun camelCaseToWords(text: String?): String? {
        if (text.isNullOrEmpty()) {
            return text
        }
        val firstLetter = text.substring(0, 1).lowercase()
        val restLetters = text.substring(1).replace("([A-Z])".toRegex(), " $1").lowercase()
        return firstLetter + restLetters
    }
}